addons/account/models/product.py

```diff
@@ -118,12 +118,9 @@ def _check_uom_not_in_invoice(self):
               JOIN product_product prod_variant ON line.product_id = prod_variant.id
               JOIN product_template prod_template ON prod_variant.product_tmpl_id = prod_template.id
               JOIN uom_uom template_uom ON prod_template.uom_id = template_uom.id
-              JOIN uom_category template_uom_cat ON template_uom.category_id = template_uom_cat.id
               JOIN uom_uom line_uom ON line.product_uom_id = line_uom.id
-              JOIN uom_category line_uom_cat ON line_uom.category_id = line_uom_cat.id
              WHERE prod_template.id IN %s
                AND line.parent_state = 'posted'
-               AND template_uom_cat.id != line_uom_cat.id
              LIMIT 1
```

Comment:
Not sure this constraint is still needed? Looks like the original fix it was added in for says it was specifically for not changing the uom category (i.e. maybe the uom can still be changed), but the validation error message say that (https://github.com/odoo/odoo/pull/110102/commits/78e4c5775d6bbb8fc7087fc94a8124406ca5da7d), so it's a bit confusing for me. I guess @odoo/rd-accounting would need to double check

Reply:
@ticodoo It seems only used for the stat buttons on the product and something catalog.  So, in itself less crucial, but it is then linked to purchase and sale, ...   I am just wondering with what you are going to replace category then?  Because you can not mix liters with kilograms, certainly in stock, while you could bring flexibility with kilograms and grams.

Reply:
@ticodoo Checking a bit further then it should be done with the reference uom...

Reply:
Removing the constrain seems like a bad idea to me.

How do you compare priced between different providers if you can buy from multiple unit categories? Which one is the cheapest? The one selling by the kg or the one selling by liters?

The right fix (to me) would be to have a conversion of mass to volume on the product for instance, but keep them separate and avoid mixing if there is no conversion available.